src/Model/TalkCommentMapper.php

```diff
@@ -235,6 +236,57 @@ protected function getBasicSQL($include_hidden = false)
         return $sql;
     }
 
+    /**
+     * @param int $user_id
+     * @throws RateLimitExceededException
+     */
+    public function checkRateLimit(int $user_id)
+    {
+        if (isset($this->_request)) {
+            $max_comments = (int) $this->_request->getConfigValue('talk_comments_rate_limit_limit', 30);
+            $duration_minutes = (int) $this->_request->getConfigValue('talk_comments_rate_limit_reset', 60);
+        } else {
+            $max_comments = 30;
+            $duration_minutes = 60;
+        }
+
+        $sql = <<<SQL
+        SELECT MIN(created_at) created_at, COUNT(ID) cnt
```

Comment:
Wouldn't this return the same result but with only one select?

```sql
SELECT min(created_at) as created_at, count(id) as cnt
FROM talk_comments
WHERE user_id = :user_id
AND created_at > :min_created_at
GROUP BY user_id
```
execute this using the `user_id` as well as `(new DateTimeImmutable())->sub(new \DateInterval(sprintf('PT%dM', $duration_minutes)))'  as `min_created_at`